cmake/modules/AddSwift.cmake

```diff
@@ -856,6 +850,31 @@ function(_add_swift_library_single target name)
     set(SWIFTLIB_SINGLE_API_NOTES "${module_name}")
   endif()
 
+  # On Linux and FreeBSD add conformance magic objects that coalesce protocol
+  # conformance sections. This wouldn't be necessary if the link was done by
+  # the swift binary: rdar://problem/19007002
+  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR
+     "${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+
+    if("${libkind}" STREQUAL "SHARED")
+      set(arch_subdir "${SWIFTLIB_DIR}/${SWIFTLIB_SINGLE_SUBDIR}")
+
+      add_custom_command_target(conformance_object_begin
```

Comment:
Thank you, it partially does.  This will definitely improve robustness, but we can do better.  Now that you have a proper CMake target that you can make other targets depend on, you don't need to execute this command (adding a target) for every library, you can do it once at the toplevel, and depend on `conformance_object_begin` everywhere.